tdrs-backend/tdpservice/parsers/validators/util.py

```diff
@@ -21,14 +21,47 @@ def make_validator(validator_func, error_func):
     def validator(value, eargs):
         try:
             if validator_func(value):
-                return (True, None)
+                return (True, None, False)
         except Exception:
             logger.exception("Caught exception in validator.")
-        return (False, error_func(eargs))
+        return (False, error_func(eargs), False)
 
     return validator
 
 
+def deprecate_validator(validator):
+    """
+    Deprecate entire validator function.
+
+    This decorator should ONLY be used on validator functions that return another
+    validator function, i.e. make_validator.
+    """
+    def wrapper(*args, **kwargs):
+        wrapper_args = args
+        wrapper_kwargs = kwargs
+
+        def deprecated_validator(*args, **kwargs):
+            make_val = validator(*wrapper_args, **wrapper_kwargs)
+            is_valid, error, _ = make_val(*args, **kwargs)
+            return (is_valid, error, True)
+        return deprecated_validator
+    return wrapper
+
+
+def deprecate_call(validator):
```

Comment:
I realized after OH 12/20/24 why this had to be a function/decorator as opposed to passing `deprecated=True` in the kwargs of the validators. The reason is because we need to modify the return value of the validator function where as the kwargs are intended to manipulate the input of the validator function. I can't see a way to get around that for the time being, but open to other suggestions! Or, if im missing something and it can be done with kwargs i'd love to pair on it!

Reply:
We also will not be able to deduce the validators true name because it has already been invoked and returns the function named `validator` from the `make_validator` function.